fix(napi/oxlint): `loadPlugin` never throw error

overlookmotel opened the pull request:
Fix an edge case bug. Ensure `loadPlugin` never throws, even if user code does something wacky like throwing `null` or `{ get message() { throw new Error("ha ha ha got you!") } }`.

camc314 commented:
### Merge activity

* **Jul 23, 2:21 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(napi/oxlint): `loadPlugin` never throw error (#12478)

Fix an edge case bug. Ensure `loadPlugin` never throws, even if user code does something wacky like throwing `null` or `{ get message() { throw new Error("ha ha ha got you!") } }`.
